### PR TITLE
Dockerise

### PR DESCRIPTION
**Pull request by RokasDevelopment**
Dockerised!

### CODE REVIEW

**Comment by biggianteye** [comment]
Builds in travis have failed.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 /vendor/
-composer.lock
 /tests/report/
+.idea/
+
```
Comment (biggianteye) [inline .gitignore]: Whitespace

File: Makefile [hunk 2]
```diff
@@ -1,28 +1,49 @@
...
 
-test-coverage:
-	@./vendor/bin/phpunit --coverage-text
+setup: ## Install dependencies and set up example conf file
+	@docker-compose run --rm composer install
+
+# Testing
+
+test: ## Run all tests
+test: test-coverage test-coverage-html test-coverage-clover test-unit test-unit-coverage test-unit-coverage-html test-unit-coverage-clover
```
Comment (biggianteye) [inline Makefile]: There's repeated build targets here. Also, you don't need to do all three versions of coverage. It's the same thing but just different output.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+version: '3'
+
+services:
+  composer:
+    image: graze/composer
+    volumes:
+      - .:/usr/src/app
+      - ~/.composer:/home/composer/.composer
+
+  php-55:
```
Comment (biggianteye) [inline docker-compose.yml]: When running the coverage tests, you'll see that it complains about xdebug not being installed. I would suggest using the `graze/php-alpine` image which is a smaller image containing some of the extra bits that we use. Also the `test` tag contains xdebug.

To maintain the kind of thing being done in this project, I would recommend this which is derived from the skeleton project.

```yaml
  test:
    image: graze/php-alpine:5-test
    volumes:
      - .:/srv
    working_dir: /srv
```

I can talk you through the changes in more detail and the reasoning behind it if you like.

**Review by biggianteye — CHANGES_REQUESTED** [review]
Looking much better. Just a few more changes.

**Review by biggianteye — APPROVED** [review]

**Comment by john-n-smith** [comment]
👏 👏 👏

**Pull request merged**